diff -rup ../packages/apps/CMUpdater/res/values/config.xml ../packages/apps/CMUpdater/res/values/config.xml
--- ../packages/apps/CMUpdater/res/values/config.xml	2017-01-14 21:18:55.553208738 +0100
+++ ../packages/apps/CMUpdater/res/values/config.xml	2017-01-14 21:18:11.000000000 +0100
@@ -8,6 +8,6 @@
 -->
 <resources xmlns:xliff="urn:oasis:names:tc:xliff:document:1.2">
     <!-- Updater Config Strings -->
-    <string name="conf_update_server_url_def" translatable="false">https://download.lineageos.org/api</string>
+    <string name="conf_update_server_url_def" translatable="false">http://aicpota.gerrett84.de/api</string>
     <string name="conf_changelog_url" translatable="false">http://localhost/changelog.xml</string>
 </resources>
diff -rup ../packages/apps/CMUpdater/src/com/cyanogenmod/updater/misc/Constants.java ../packages/apps/CMUpdater/src/com/cyanogenmod/updater/misc/Constants.java
--- ../packages/apps/CMUpdater/src/com/cyanogenmod/updater/misc/Constants.java	2017-01-14 21:18:55.553208738 +0100
+++ ../packages/apps/CMUpdater/src/com/cyanogenmod/updater/misc/Constants.java	2017-01-14 21:12:46.000000000 +0100
@@ -11,7 +11,7 @@ package com.cyanogenmod.updater.misc;
 
 public class Constants {
     // Download related
-    public static final String UPDATES_FOLDER = "updates";
+    public static final String UPDATES_FOLDER = "aicpupdater";
     public static final String DOWNLOAD_ID = "download_id";
 
     // Preferences
@@ -35,7 +35,7 @@ public class Constants {
     public static final int UPDATE_TYPE_NIGHTLY = 1;
 
     // ro.cm.releasetype values
-    public static final String PROPERTY_CM_RELEASETYPE = "ro.cm.releasetype";
+    public static final String PROPERTY_CM_RELEASETYPE = "ro.romstats.buildtype";
     public static final String CM_RELEASETYPE_SNAPSHOT = "SNAPSHOT";
     public static final String CM_RELEASETYPE_NIGHTLY = "NIGHTLY";
 }
diff -rup ../packages/apps/CMUpdater/src/com/cyanogenmod/updater/UpdatesSettings.java ../packages/apps/CMUpdater/src/com/cyanogenmod/updater/UpdatesSettings.java
--- ../packages/apps/CMUpdater/src/com/cyanogenmod/updater/UpdatesSettings.java	2017-01-14 21:18:55.553208738 +0100
+++ ../packages/apps/CMUpdater/src/com/cyanogenmod/updater/UpdatesSettings.java	2017-01-14 21:14:17.000000000 +0100
@@ -596,7 +596,7 @@ public class UpdatesSettings extends Pre
         mUpdatesList.removeAll();
 
         // Convert the installed version name to the associated filename
-        String installedZip = "cm-" + Utils.getInstalledVersion() + ".zip";
+        String installedZip = "aicp-" + Utils.getInstalledVersion() + ".zip";
 
         // Convert LinkedList to HashMap, keyed on filename.
         HashMap<String, UpdateInfo> updatesMap = new HashMap<String, UpdateInfo>();
diff -rup ../packages/apps/CMUpdater/src/com/cyanogenmod/updater/utils/Utils.java ../packages/apps/CMUpdater/src/com/cyanogenmod/updater/utils/Utils.java
--- ../packages/apps/CMUpdater/src/com/cyanogenmod/updater/utils/Utils.java	2017-01-14 21:18:55.553208738 +0100
+++ ../packages/apps/CMUpdater/src/com/cyanogenmod/updater/utils/Utils.java	2017-01-14 21:16:29.000000000 +0100
@@ -54,11 +54,11 @@ public class Utils {
     }
 
     public static String getDeviceType() {
-        return SystemProperties.get("ro.cm.device");
+        return SystemProperties.get("ro.aicp.device");
     }
 
     public static String getInstalledVersion() {
-        return SystemProperties.get("ro.cm.version");
+        return SystemProperties.get("ro.modversion");
     }
 
     public static int getInstalledApiLevel() {
